src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateParser.java

```diff
@@ -0,0 +1,202 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.swing.text.html.HTML.Attribute;
+
+import org.apache.maven.doxia.macro.MacroExecutionException;
+import org.apache.maven.doxia.macro.MacroRequest;
+import org.apache.maven.doxia.macro.manager.MacroNotFoundException;
+import org.apache.maven.doxia.module.xdoc.XdocParser;
+import org.apache.maven.doxia.parser.ParseException;
+import org.apache.maven.doxia.parser.Parser;
+import org.apache.maven.doxia.sink.Sink;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.util.IOUtil;
+import org.codehaus.plexus.util.xml.pull.XmlPullParser;
+
+/**
+ * Parser for Checkstyle's xdoc templates.
+ * This parser is responsible for generating xdocs({@code .xml}) from the xdoc
+ * templates({@code .xml.template}). The templates are regular xdocs with custom
+ * macros for generating dynamic content - properties, examples, etc.
+ * This parser behaves just like the {@link XdocParser} with the difference that all
+ * elements apart from the {@code macro} element are copied as is to the output.
+ *
+ * @see ExampleMacro
+ */
+@Component(role = Parser.class, hint = "xdocs-template")
+public class XdocsTemplateParser extends XdocParser {
```

Comment:
Just wanted to confirm, this class is only temporary while we keep the template files and when we rename `xml.template` to `xml` this will go away? Are there any other classes here that are like this?

Reply:
Template files will be be forever with us, they are source of  html layout

Reply:
Yes, once all files are moved to a template with macros, this file along with `XdocsTemplateSink` and `XdocsTemplateSinkFactory` can go away. The only reason we have them now is to use a custom parser that executes the macros and the rest of the xdoc stays the same. Once we ensure that generation is okay, we can drop those - the default `XdocParser` will both translate the macros and the rest of the content to html.

Reply:
Ok, we need an issue to cover that and add all the temporary files/tests that will eventually be removed. It will also have to be tied to the metadata issue and gutting the javadoc. Each of the class should have an until that issue number in the code so CI picks it up.

Reply:
I moved task `Xdoc generation should be in mvn site phase` to an issue and wrote a description(https://github.com/checkstyle/checkstyle/issues/13426). This is the moment when these modules will be dropped. Added links to `XdocsTemplateParser` `XdocsTemplateSink`, `XdocsTemplateSinkFactory`, `XdocGenerator` and `XdocsPagesTest::generateXdocContent()`.